File: stepik-java-api/src/main/java/org/stepik/api/objects/steps/Steps.java
package org.stepik.api.objects.steps;

import org.jetbrains.annotations.NotNull;
import org.stepik.api.objects.ObjectsContainer;

import java.util.ArrayList;
import java.util.List;

/**
 * @author meanmail
 */
public class Steps extends ObjectsContainer<Step> {
    private List<Step> steps;

    @NotNull
    public List<Step> getSteps() {
        if (steps == null) {
            steps = new ArrayList<>();
        }
        return steps;
    }

    @NotNull
    @Override
    public List<Step> getItems() {
        return getSteps();
    }

    @NotNull
    @Override
    public Class<Step> getItemClass() {
        return Step.class;
    }
}

File: stepik-java-api/src/main/java/org/stepik/api/objects/steps/Steps.kt
package org.stepik.api.objects.steps

import com.google.gson.annotations.JsonAdapter
import com.google.gson.annotations.SerializedName
import org.stepik.api.client.serialization.DefaultAsEmptyLessonArrayAdapter
import org.stepik.api.client.serialization.DefaultAsEmptyStepArrayAdapter
import org.stepik.api.objects.ObjectsContainer


class Steps : ObjectsContainer<Step>() {
    @SerializedName("steps")
    @JsonAdapter(DefaultAsEmptyStepArrayAdapter::class, nullSafe = false)
    override var items = mutableListOf<Step>()

    override val itemClass
        get() = Step::class.java
}
